spec/lib/message_bus_spec.rb

```diff
@@ -216,6 +196,40 @@
     @bus.backlog("/chuck").map { |i| i.data }.to_a.must_equal ['foo']
   end
 
+  it "can be turned off" do
+    @bus.off
+
+    @bus.off?.must_equal true
+
+    @bus.publish("/chuck", "norris")
+
+    @bus.backlog("/chuck").to_a.must_equal []
+  end
+
+  it "can be turned off only for subscriptions" do
+    @bus.off(disable_publish: false)
+
+    @bus.off?.must_equal true
+
+    @bus.publish("/chuck", "norris")
+
+    @bus.backlog("/chuck").map(&:data).to_a.must_equal ["norris"]
```

Comment:
this is close but a bit confusing, you should be attempting a new subscription and that should fail, seems to only be testing part of the change?

Reply:
If I do an explicit subscription after calling `#off`, the subscription works. `#off` only affects the implicit middleware subscription.

Reply:
What do you think of https://github.com/discourse/message_bus/pull/317/commits/45fd7387230e705938fa74e901af235c15fc1f64 ?